File: src/tests/test_parser/GettersAndSettersPruning.py
from JavaParserVisitor import ClassInfo, MethodInfo

def expectedAnswer():
    ans = dict()
    test = ClassInfo("Test")
    test.members.add("x")
    test.members.add("y")

    test.methods["foo"] = MethodInfo("foo")

    getZ = MethodInfo("getZ")
    getZ.memberReferences.add("y")
    test.methods["getZ"] = getZ

    ans["Test"] = test
    
    return ans

File: src/tests/test_parser/GettersAndSettersPruning.java
public class Test {
  void getX() {
    return x;
  }
  void getx() {
    return x;
  }
  void setX(int z) {
    x = z;
  }
  void setx(int z) {
    x = z;
  }
  void foo() {
    getX();
    getx();
    setX();
    setx();
  }  
  void getZ() {
    y += 1;
  }
  int x;
  int y;
}